### PR TITLE
Progressive web app

### PR DESCRIPTION
**Pull request by mshafer1**
Add manifest and service worker to turn this website into a Progressive Web App (PWA) - with offline functionality through caching.

This adds an "Add to Home Screen" behavior that makes the website open like a native app - even when in airplane mode.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (mshafer1) [inline _config.yml:1]: This file is a Jekyll config - it allows for defining variables (strings) that are used throughout the site.

File: manifest.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+            "src": "{{site.logo}}",
+            "sizes": "560x560",
+            "type": "image/png"
+        }
+    ],
+    "start_url": "/",
+    "display": "standalone",
+    "background_color": "#800000",
+    "theme_color": "#ffd700",
+    "timestamp": "{{ site.time | date: "%Y-%m-%d %H:%M" | url_encode }}"
```
Comment (mshafer1) [inline manifest.json:17]: this is an arbitrary value to cause people with caches to update when a new version of the site is available (as the timestamp will cause this file to show as updated).

File: service-worker.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  'scripts/jquery.js',
+  'scripts/seasonConfigScript.js',
+  'scripts/eventDataScript.js',
+  'scripts/teamDataScript.js',
+  'scripts/importing-Exporting-Config.js',
+  'scripts/commonMethodHelpers.js',
+  'scripts/autosize-master/dist/autosize.js',
+  '{{site.logo}}',
+  'images/background.jpg',
+  'manifest.json',
```
Comment (mshafer1) [inline service-worker.js:23]: I tried to list every file here, we should add a backlog item to use Jekyll to generate this.

**Review by mshafer1 — COMMENTED** [review]
I tested this with the "install" option in Chrome on my PC - while pointing my phone at my PC gave me an "add to home screen" behavior, it doesn't work in offline mode. I **think** this is because I'm not serving over http. We'll be able to test this once we get this PR in.

**Review by harishbommakanti — APPROVED** [review]
The 'install feature' on chrome works but we'll have to do the PR to actually resolve any conflicts that may rise when using the web app on mobile. Looks good!

**Pull request merged**